pkg/operator/csi/csidrivercontrollerservicecontroller/helpers.go

```diff
@@ -190,8 +190,43 @@ func WithPlaceholdersHook(configInformer configinformers.SharedInformerFactory)
 		logLevel := loglevel.LogLevelToVerbosity(spec.LogLevel)
 		pairs = append(pairs, []string{"${LOG_LEVEL}", strconv.Itoa(logLevel)}...)
 
-		// Default TLS Cipher Suites for Kube RBAC sidecars
-		pairs = append(pairs, []string{"${TLS_CIPHER_SUITES}", defaultTLSCipherSuites}...)
+		replaced := strings.NewReplacer(pairs...).Replace(string(manifest))
+		return []byte(replaced), nil
+	}
+}
+
+// WithServingInfo is a manifest hook that replaces ${TLS_CIPHER_SUITES} and ${TLS_MIN_VERSION}
+// placeholders with the observed configuration.
+func WithServingInfo() dc.ManifestHookFunc {
+	return func(opSpec *opv1.OperatorSpec, manifest []byte) ([]byte, error) {
+		if len(opSpec.ObservedConfig.Raw) == 0 {
+			return manifest, nil
+		}
+
+		var config map[string]interface{}
+		if err := json.Unmarshal(opSpec.ObservedConfig.Raw, &config); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal the observedConfig: %w", err)
+		}
+
+		cipherSuites, cipherSuitesFound, err := unstructured.NestedStringSlice(config, csiconfigobservercontroller.CipherSuitesPath()...)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't get the servingInfo.cipherSuites config from observed config: %w", err)
+		}
+
+		minTLSVersion, minTLSVersionFound, err := unstructured.NestedString(config, csiconfigobservercontroller.MinTLSVersionPath()...)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't get the servingInfo.minTLSVersion config from observed config: %v", err)
+		}
+
+		pairs := []string{}
+		if cipherSuitesFound && len(cipherSuites) > 0 {
+			pairs = append(pairs, []string{"${TLS_CIPHER_SUITES}", strings.Join(cipherSuites, ",")}...)
+
+		}
+
+		if minTLSVersionFound && len(minTLSVersion) > 0 {
```

Comment:
It would be good to mention that we expect `minTLSVersion` to be always present, because all TLS profiles specify it + `apiserver.spec.tlsSecurityProfile.custom.minTLSVersion` [cannot be empty](https://github.com/openshift/api/blob/c90b4c82852ead1fdb58866b685fe75b0b62163d/config/v1/0000_10_config-operator_01_apiserver-Default.crd.yaml#L144-L151).

Similarly for `cipherSuites` - validation prevents empty `spec.tlsSecurityProfile.custom.ciphers`